src/arfsdao_anonymous.ts

```diff
@@ -46,14 +46,18 @@ export class ArFSDAOAnonymous extends ArFSDAOType {
 	}
 
 	public async getOwnerForPublicDriveId(driveId: DriveID): Promise<ArweaveAddress> {
-		return this.getOwnerForDriveId(driveId, 'public');
+		return this.getOwnerForDriveId(driveId, true);
 	}
 
-	public async getOwnerForPrivateDriveId(driveId: DriveID): Promise<ArweaveAddress> {
-		return this.getOwnerForDriveId(driveId, 'private');
+	public async getOwnerForPrivateDriveId(driveId: DriveID, driveKey: DriveKey): Promise<ArweaveAddress> {
+		return this.getOwnerForDriveId(driveId, true, driveKey);
 	}
 
-	public async getOwnerForDriveId(driveId: DriveID, drivePrivacy?: DrivePrivacy): Promise<ArweaveAddress> {
+	public async getOwnerForDriveId(
+		driveId: DriveID,
+		assertPrivacy = false,
+		driveKey?: DriveKey
```

Comment:
So unfortunately this violates the "anonymous" part of of this class. We need to hoist the private versions of this function into the ArFSDAO layer.

Reply:
I've addressed this here: b81960c

Let me know what you think!